server/store/sqlstore/migrations.go

```diff
@@ -17,6 +17,18 @@ func (s *SQLStore) runMigrationRemoteID(remoteID string) error {
 	return err
 }
 
+func (s *SQLStore) runSetEmailVerifiedToTrueForRemoteUsers(remoteID string) error {
+	_, err := s.getQueryBuilder().
```

Comment:
Changing the code locally  to get the SQL query if you are interested:

```sql
UPDATE Users SET EmailVerified = $1 WHERE (RemoteID = $2 AND EmailVerified = $3)
```

```
ARGS=[true pp9fktok8bfr8bmztwkyerifoh false]
```

note:remoteID will be different for each installation so yours will differ

Reply:
Is there any way we can test the performance of this query against a large `Users` table.

Reply:
I'm adding 30K users to my local env right now, not sure if there's better way! If perfs are good, we'll ship my mac to production :D

Reply:
Status update:

With 9628 users, it takes ~730ms locally.

```jsonl
{"timestamp":"2024-04-11 15:23:53.975 -07:00","level":"debug","msg":"Starting runSetEmailVerifiedToTrueForRemoteUsers","caller":"app/plugin_api.go:1001","plugin_id":"com.mattermost.msteams-sync"}
{"timestamp":"2024-04-11 15:23:54.705 -07:00","level":"debug","msg":"Finished runSetEmailVerifiedToTrueForRemoteUsers","caller":"app/plugin_api.go:1001","plugin_id":"com.mattermost.msteams-sync","duration":"729.708167ms","updated_users":"9628"}
```

Still waiting to get my 30K users ready!

Reply:
Thanks for doing this...make me feel better.

Reply:
15K users => 1.134357s

```
{"timestamp":"2024-04-11 17:21:16.516 -07:00","level":"debug","msg":"Starting runSetEmailVerifiedToTrueForRemoteUsers","caller":"app/plugin_api.go:1001","plugin_id":"com.mattermost.msteams-sync"}
{"timestamp":"2024-04-11 17:21:17.650 -07:00","level":"debug","msg":"Finished runSetEmailVerifiedToTrueForRemoteUsers","caller":"app/plugin_api.go:1001","plugin_id":"com.mattermost.msteams-sync","duration":"1.134357s","updated_users":"15032"}
```

Reply:
So double or triple that and we are at 2.5 - 3 seconds. Which I think is ok for this, at least its not 30. But as Caleb mentioned below, we have to be careful because right now, these will run every time the plugin starts. We ran into some long migration queries in Boards.

I'm ok merging and moving forward.

Reply:
Final report here, with 27K users it took 2.1sec so it does not increase exponentially. 

Restarting the plugin a few times in a row with the 27K users gives me 2.1sec the first time (because we update all the users the first time), and all following restarts of the plugin take ~50ms with 0 users affected. So I don't see how the migration can grow much slower, unless we somehow insert a ton of synthetic user with EmailVerified=false... which should not happen anymore I reckon.

Reply:
Awesome. I say we move forward, merge and cherry-pick. Then I can start the build. Thanks.